### PR TITLE
support custom coffeescript forks (fix #52)

### LINKED ISSUE #52: ES6 support
Any plans on supporting ES6? Similarly to [this fork](https://github.com/t3chnoboy/gulp-coffee-es6)..

### PR DESCRIPTION
**Pull request by bcherny**

### CODE REVIEW

**Comment by bcherny** [comment]
Tests seem to have been failing already :(

File: index.js [hunk 1]
```diff
@@ -23,6 +22,7 @@ module.exports = function (opt) {
 
     var options = merge({
       bare: false,
+      coffee: require('coffeescript'),
```
Comment (contra) [inline index.js]: typo

Comment (bcherny) [inline index.js]: That's embarrassing... and explains the test failures. Good catch. I was using it to test coffeescript6.

File: index.js [hunk 2]
```diff
@@ -23,6 +22,7 @@ module.exports = function (opt) {
 
     var options = merge({
       bare: false,
+      coffee: require('coffee-script'),
```
Comment (contra) [inline index.js]: shouldnt we lazy load this?

Comment (bcherny) [inline index.js]: We should if (1) the cost of parsing+executing coffee-script is high, or (2) requiring coffee-script has side effects. I don't think either of these are the case.

What do you think?

Comment (contra) [inline index.js]: @bcherny Ah I thought we had it as a peer dep, nvm

**Comment by contra** [comment]
Needs a docs and a test if possible

**Comment by bcherny** [comment]
Done.

**Comment by contra** [comment]
Published as 2.3.4

**Comment by bcherny** [comment]
BTW, you might want to rm Node <4 from the travis.yml. Seems to be causing tests to fail.

**Pull request merged**